docs/kvrocks-search.md

```diff
@@ -0,0 +1,269 @@
+# Search
+
+**Apache Kvrocks™** Search, also known as **Kvrocks Search** (or KQIR, as a technical term), is an internal component of Apache Kvrocks™. It functions as a query engine that supports (secondary) indexing on structured data and complex queries by effectively utilizing various indexes.
+
+In addition to being compatible with many commands and the query syntax of [RediSearch](https://redis.io/docs/latest/develop/interact/search-and-query/) (e.g. [FT.CREATE](#ftcreate) and [FT.SEARCH](#ftsearch)), Kvrocks Search also offers support for SQL syntax to accommodate various scenarios (via [FT.SEARCHSQL](#ftsearchsql-extension) and other related commands).
```

Comment:
So, the extension is "extension on RediSearch syntax"?

Reply:
No. It's extension on RediSearch commands. SQL and RediSearch query syntax has no connection.